newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudResourceMover.js

```diff
@@ -161,15 +162,113 @@ export const moveAllCloudProjectResourcesToCloudProject = async ({
   uploadedProjectResourceFiles.forEach(({ url, error }, index) => {
     const resource = downloadedFilesAndResourcesToUpload[index].resource;
     if (error || !url) {
-      result.erroredResources.push({
-        resourceName: resource.getName(),
-        error: error || new Error('Unknown error during upload.'),
-      });
+      onError(
+        resource.getName(),
+        error || new Error('Unknown error during upload.')
+      );
       return;
     }
 
     resource.setFile(url);
   });
+};
+
+export const moveAllCloudProjectResourcesToCloudProject = async ({
+  project,
+  authenticatedUser,
+  oldFileMetadata,
+  newFileMetadata,
+  oldStorageProvider,
+  oldStorageProviderOperations,
+  newStorageProvider,
+  newStorageProviderOperations,
+  onProgress,
+}: MoveAllProjectResourcesOptions): Promise<MoveAllProjectResourcesResult> => {
+  const result: MoveAllProjectResourcesResult = {
+    erroredResources: [],
+  };
+
+  const newCloudProjectId = newFileMetadata.fileIdentifier;
+
+  const resourcesToFetchAndUpload = getResourcesToFetchAndUpload({
+    project,
+    selectCondition: checkIfIsGDevelopCloudBucketUrl,
+    ignoreCondition: fileName =>
+      extractProjectUuidFromProjetResourceUrl(fileName) === newCloudProjectId,
+    onError: (resourceName, errorMessage) => {
+      result.erroredResources.push({
+        resourceName,
+        error: new Error(errorMessage),
+      });
+    },
+  });
+
+  // If an error happens here, it will be thrown out of the function.
+  if (oldStorageProviderOperations.onEnsureCanAccessResources)
+    oldStorageProviderOperations.onEnsureCanAccessResources(
+      project,
+      oldFileMetadata
+    );
+
+  await uploadResourcesToCloud({
+    resourcesToFetchAndUpload,
+    authenticatedUser,
+    cloudProjectId: newCloudProjectId,
+    onProgress,
+    onError: (resourceName, error) => {
+      result.erroredResources.push({
+        resourceName,
+        error,
+      });
+    },
+  });
+
+  return result;
+};
+
+export const moveUrlResourcesToCloudFilesIfPrivate = async ({
+  project,
+  fileMetadata,
+  authenticatedUser,
+  onProgress,
+}: {|
+  project: gdProject,
+  fileMetadata: FileMetadata,
+  authenticatedUser: AuthenticatedUser,
+  onProgress: (number, number) => void,
+|}) => {
+  const result = {
+    erroredResources: [],
+  };
+
+  const cloudProjectId = fileMetadata.fileIdentifier;
+
+  // Get all resources to download.
+  const resourcesToFetchAndUpload = getResourcesToFetchAndUpload({
+    project,
+    selectCondition: isPrivateAssetUrl,
+    onError: (resourceName, errorMessage) => {
+      result.erroredResources.push({
+        resourceName,
+        error: new Error(errorMessage),
+      });
+    },
+  });
+
+  await uploadResourcesToCloud({
+    resourcesToFetchAndUpload,
+    // // Remove the token used to download the file when re-uploading it.
+    transformFilename: filename => filename.split('?')[0],
```

Comment:
This is a very important thing, that is specific to the private assets. It definitely deserves a named function to explain what it does + a test.